### PR TITLE
Integrate Keycloak Client #7

### PR DESCRIPTION
**Pull request by mjwdavis**
https://github.com/orgs/UCDavisLibrary/projects/196/views/1?pane=issue&itemId=59466753

### CODE REVIEW

File: src/lib/serverConfig.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ class ServerConfig {
     this.env = process?.env?.APP_ENV === 'dev' ? 'dev' : 'prod';
 
     // TODO: Replace these with your own app title
-    this.title = this.getEnv('APP_TITLE', 'UC Davis Library Simple SPA');
+    this.title = this.getEnv('APP_TITLE', 'UC Davis Library Travel App');
```
Comment (spelkey-ucd) [inline src/lib/serverConfig.js:12]: I have a separate issue for this. Debbie had a different name in mind.

File: src/lib/utils/AccessToken.js [hunk 2]
```diff
@@ -34,6 +34,13 @@ export default class AccessToken {
     return this._inRoleList('admin-access');
   }
 
+  /**
+   * @description Returns true if user has report access to this client
+   */
+    get hasReportAccess(){
```
Comment (spelkey-ucd) [inline src/lib/utils/AccessToken.js:40]: we also want admin to be able to access reports

File: src/lib/utils/AccessToken.js [hunk 3]
```diff
@@ -34,6 +34,13 @@ export default class AccessToken {
...
+      return this._inRoleList('report-access');
```
Comment (spelkey-ucd) [inline src/lib/utils/AccessToken.js:41]: This would also grant access to the `report-access` realm role - we only want client.

**Review by spelkey-ucd — CHANGES_REQUESTED** [review]
Keycloak config looks good except you don't want "Client Authentication" checked (I unchecked it). That is only for auth flows that require a client secret. In SPAs, we do our auth on the browser-side which then passes the access token to the server in the API requests (check out the [client-side BaseService](https://github.com/UCDavisLibrary/ucdlib-travel/blob/main/src/lib/cork/services/BaseService.js#L28) and then [the API middleware](https://github.com/UCDavisLibrary/ucdlib-travel/blob/main/src/api/auth.js#L9) ). Client secrets are only for straight up server-side auth, otherwise the secret would be leaked in the browser.

image

File: src/lib/utils/AccessToken.js (hunk 2 above)
Comment (mjwdavis) [reply src/lib/utils/AccessToken.js:40]: Added roles as array to hasAdminAccess.

File: src/lib/utils/AccessToken.js (hunk 3 above)
Comment (mjwdavis) [reply src/lib/utils/AccessToken.js:41]: removed function

File: src/lib/serverConfig.js (hunk 1 above)
Comment (mjwdavis) [reply src/lib/serverConfig.js:12]: Updated title to "UC Davis Library Travel, Training, and Professional Development" per Figma design

**Review by spelkey-ucd — CHANGES_REQUESTED** [review]

**Pull request merged**